### PR TITLE
[CODEGEN][OPENCL] Fix compile error about ternary expression.

### PR DESCRIPTION
**Pull request by lixiaoquan**
Code like this can't be built with NV OpenCL, and it needs an explicit type
  converison for ternary expression if return type is uchar.

       uchar i = 0, j = 0;
       uchar t = max((uchar)j, ((i > 0) ? (uchar)1 : (uchar)0));

It is for #2787

### CODE REVIEW

File: src/codegen/codegen_opencl.cc [hunk 1]
```diff
@@ -208,6 +208,16 @@ void CodeGenOpenCL::VisitExpr_(const Broadcast* op, std::ostream& os) {   // NOL
   os << "))";
 }
 
+void CodeGenOpenCL::VisitExpr_(const Call *op, std::ostream& os) {  // NOLINT(*)
+  if (op->is_intrinsic(intrinsic::tvm_if_then_else)) {
```
Comment (tqchen) [inline src/codegen/codegen_opencl.cc]: Please add a comment on why explicit cast is needed here.

Likely you also need to add support for select

Comment (tqchen) [reply src/codegen/codegen_opencl.cc]: Please add a regression test case as per https://docs.tvm.ai/contribute/code_review.html#ensure-test-coverage

File: src/codegen/codegen_opencl.cc [hunk 2]
```diff
@@ -208,6 +208,16 @@ void CodeGenOpenCL::VisitExpr_(const Broadcast* op, std::ostream& os) {   // NOL
...
+    if (op->args[2].type() == UInt(8)) {
```
Comment (kazum) [inline src/codegen/codegen_opencl.cc]: On my environment, the problem also happens with char, short, and ushort.  It looks safe to add a cast for any types.

Comment (lixiaoquan) [reply src/codegen/codegen_opencl.cc]: It works with float/int, that's why most people didn't run into this problem. Casting for any types is good for me.

File: src/codegen/codegen_opencl.cc (hunk 1 above)
Comment (lixiaoquan) [reply src/codegen/codegen_opencl.cc]: Tests added for both tvm::if_then_else and Select

**Comment by kazum** [comment]
@tqchen I think your comment has been addressed.  Can you confirm it and update your review status if appropriate?

**Comment by kazum** [comment]
Thanks @lixiaoquan @tqchen, this is now merged.

**Pull request merged**